.circleci/config.yml

```
@@ -1,111 +1,111 @@
+                            name: upload coverage report
+                            command: |
+                                npx codecov -f packages/core/coverage/lcov.info -F core
+                                npx codecov -f packages/http-server/coverage/lcov.info -F http-server
+                                npx codecov -f packages/http-server-express/coverage/lcov.info -F http-server-express
+                                npx codecov -f packages/reflector/coverage/lcov.info -F reflector
+                      - save_cache:
+                            key: v1-source-{{ .Environment.CIRCLE_SHA1 }}
+                            paths:
+                                - '~'
+    publish:
+        docker:
+            - image: cimg/node:20.11.0
+        steps:
+            - restore_cache:
+                  key: v1-source-{{ .Environment.CIRCLE_SHA1 }}
+            - add_ssh_keys
+            - run:
+                  name: Add github to known_hosts
+                  command: |
+                      mkdir -p ~/.ssh
+                      ssh-keyscan github.com >> ~/.ssh/known_hosts
+            - run:
+                  name: publish
+                  command: |
+                      git config --global user.email "circleci@hp.com"
+                      git config --global user.name "CircleCI"
+                      git checkout -f
+                      npm run semantic-release
 
-  publish_docs:
-    docker:
-      - image: cimg/node:16.18.1
-    steps:
-      - restore_cache:
-          key: v1-source-{{ .Environment.CIRCLE_SHA1 }}
-      - add_ssh_keys
-      - run:
-          name: Deploy docs
-          command: |
-            rm .npmrc
-            sh ./docs/deploy.sh
+    publish_docs:
+        docker:
+            - image: cimg/node:20.11.0
+        steps:
+            - restore_cache:
+                  key: v1-source-{{ .Environment.CIRCLE_SHA1 }}
+            - add_ssh_keys
+            - run:
+                  name: Deploy docs
+                  command: |
+                      rm .npmrc
+                      sh ./docs/deploy.sh
 
 workflows:
-  version: 2
-  build_and_publish:
-    jobs:
-      - checkout_and_test:
-          matrix:
-            parameters:
-              mongo-version: ["4.4", "5"]
-              docker-image: ["circleci/node:14", "circleci/node:16", "cimg/node:18.16.0"]
-      - publish:
-          requires:
-            - checkout_and_test
-          filters:
-            branches:
-              only:
-                - master
-      - publish_docs:
-          requires:
-            - checkout_and_test
-          filters:
-            branches:
-              only:
-                - master
+    version: 2
+    build_and_publish:
+        jobs:
+            - checkout_and_test:
+                  matrix:
+                      parameters:
+                          mongo-version: ['5', '6']
+                          docker-image: ['cimg/node:18.19.0', 'cimg/node:20.11.0']
```

Do we need to pin the exact versions for the testing? I see some benefits with pulling the latest major releases here

```suggestion
                          docker-image: ['cimg/node:18', 'cimg/node:20']
```

---

seems like it doesn't work. it needs the exact image tag to pull